extensions/sql-database-projects/src/controllers/projectController.ts

```diff
@@ -32,11 +45,138 @@ export class ProjectsController {
 
 		// Read datasources.json (if present)
 		const dataSourcesFilePath = path.join(path.dirname(projectFile.fsPath), constants.dataSourcesFileName);
-		newProject.dataSources = await dataSources.load(dataSourcesFilePath);
+
+		try {
+			newProject.dataSources = await dataSources.load(dataSourcesFilePath);
+		}
+		catch (err) {
+			if (err instanceof dataSources.NoDataSourcesFileError) {
+				// TODO: prompt to create new datasources.json; for now, swallow
+			}
+			else {
+				throw err;
+			}
+		}
+
+		this.refreshProjectsTree();
+	}
+
+	public async createNewProject(newProjName: string, newProjUri: vscode.Uri) {
+		const macroDict: Record<string, string> = {
+			'PROJECT_NAME': newProjName,
+			'PROJECT_GUID': '00000000-0000-0000-0000-000000000000'//Guid.create().toString() // TODO: extension building problems when using this library?
```

Comment:
I'm using this library to generate guid: 
import * as UUID from 'vscode-languageclient/lib/utils/uuid';